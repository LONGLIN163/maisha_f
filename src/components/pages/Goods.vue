<template>
    <div class="goodsbox">
        <div class="navbar-div">
            <van-nav-bar
                title="Product Details"
                left-text="Back"
                left-arrow
                @click-left="onClickLeft"
            ></van-nav-bar>
            <!--***van-nav-bar has to be a close tag, otherwise will have error occurs in testing***-->
        </div>
        <div class="topimage-div">
            <img :src="goodsInfo.IMAGE1" width="100%"/>
        </div>
        <div class="goods-name">{{goodsInfo.NAME}}</div>
        <div class="goods-price">Price:{{goodsInfo.PRESENT_PRICE | moneyFilter}}â‚¬</div>
        <div>
            <van-tabs swipeable sticky>
                <van-tab  title="Details">
                    <div class="detail" v-html="goodsInfo.DETAIL">

                    </div>
                </van-tab>
                <van-tab title="Comment">no any comments...</van-tab>
            </van-tabs>
        </div>
        <div class="goods-bottom">
            <div>
                <van-button data-test="addtocard" size="large" type="primary" @click="addGoodsToCart">ADD TO CART</van-button>
            </div>
            <div>
                <van-button size="large" type="danger">Buy</van-button>
            </div>
        </div>
    </div>
</template>

<script>
    import axios from 'axios'
    import serviceApi from '@/serviceApi.config.js'
    import {Toast} from 'vant'
    import {toMoney} from '@/filters/moneyFilter.js'

    export default {
        data() {
            return {
                goodsId: '',
                goodsInfo:{}, 
            }
        },
        created(){
            this.goodsId= this.$route.query.goodsId ? this.$route.query.goodsId:this.$route.params.goodsId
            this.getInfo()
        },
        methods: {
            getInfo() {
                axios({
                    url:serviceApi.getDetailGoodsInfo,
                    method:'post',
                    data:{
                        goodsId:this.goodsId
                    }
                })
                .then(res=>{
                    // before we assign goodsInfo, we need to make some judgments 
                    if(res.data.code == 200 && res.data.message ){
                            this.goodsInfo = res.data.message 
                    }else{
                        //Toast('Server error, fetching data failed!!!')
                    }
                    console.log( this.goodsInfo)
                })
                .catch(error=>{
                    console.log(error)
                })
            },
            onClickLeft(){ // go to last page
                this.$router.go(-1)
            },
            async addGoodsToCart(){
                //extract caretInfo for local first
                let cartInfo = localStorage.cartInfo ? JSON.parse(localStorage.cartInfo) :[]
                //check if target product is already exist
                let isHaveGoods=await cartInfo.find(item=>item.goodsId==this.goodsId)
                if(!isHaveGoods){ // add this product to the cart
                    let newGoodsInfo={
                        goodsId:this.goodsInfo.ID,
                        Name:this.goodsInfo.NAME,
                        price:this.goodsInfo.PRESENT_PRICE,
                        image:this.goodsInfo.IMAGE1,
                        count:1
                    }
                    cartInfo.push(newGoodsInfo) 
                    localStorage.cartInfo=JSON.stringify(cartInfo) //save to local
                    Toast.success('add to cart success!')

                }else{
                    Toast.success('this product is already exist!')
                }
                this.$router.push({name:'Cart'}) 
            }
        },
        filters:{
            moneyFilter(money){
                return toMoney(money)
            }  
        },
    }
</script>

<style lang="scss" scoped>
   .detail{
     font-size:0px;
    }
    .goods-name{
        background-color: $color-background-sub;
    }
    .goods-price{
        background-color: $color-background-sub;
    }
    .goods-bottom{
        position: fixed;
        bottom:0px;
        left:0px;
        background-color: $color-background-sub;
        width:100%;

        display: flex;
        flex-direction: row;
        flex-flow: nowrap;
        div{
            flex:1;
            padding:2px;
        }
    }
</style>