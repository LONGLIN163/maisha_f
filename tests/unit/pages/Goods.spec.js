import Vue from 'vue'
import { shallowMount,mount, createLocalVue} from '@vue/test-utils'

import Goods from '@/components/pages/Goods.vue'

import VueRouter from 'vue-router';
const Vant=require("vant")

const localVue = createLocalVue();
localVue.use(VueRouter)
localVue.use(Vant); 

const router = new VueRouter();


describe('Goods', () => {
  const wrapper = mount(Goods,{
    localVue,
    router
  })

  it('renders Goods', () => {
    expect(wrapper).toMatchSnapshot()
  })
})