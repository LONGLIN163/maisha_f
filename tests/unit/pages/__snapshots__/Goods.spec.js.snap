// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goods renders Goods 1`] = `
<div class="goodsbox">
  <div class="navbar-div">
    <div class="van-nav-bar van-hairline--bottom">
      <div class="van-nav-bar__content">
        <div class="van-nav-bar__left"><i class="van-icon van-icon-arrow-left van-nav-bar__arrow">
            <!---->
          </i><span class="van-nav-bar__text">Back</span></div>
        <div class="van-nav-bar__title van-ellipsis">Product Details</div>
      </div>
    </div>
  </div>
  <div class="topimage-div"><img width="100%"></div>
  <div class="goods-name"></div>
  <div class="goods-price">Price:0.00â‚¬</div>
  <div>
    <div class="van-tabs van-tabs--line">
      <div>
        <div class="van-sticky">
          <div class="van-tabs__wrap">
            <div role="tablist" class="van-tabs__nav van-tabs__nav--line">
              <div role="tab" aria-selected="true" class="van-tab van-tab--active"><span class="van-tab__text van-tab__text--ellipsis">Details</span></div>
              <div role="tab" class="van-tab"><span class="van-tab__text van-tab__text--ellipsis">Comment</span></div>
              <div class="van-tabs__line"></div>
            </div>
          </div>
        </div>
      </div>
      <div class="van-tabs__content">
        <div role="tabpanel" class="van-tab__pane" style="">
          <div class="detail"></div>
        </div>
        <div role="tabpanel" class="van-tab__pane" style="display: none;">
          <!---->
        </div>
      </div>
    </div>
  </div>
  <div class="goods-bottom">
    <div><button class="van-button van-button--primary van-button--large">
        <div class="van-button__content"><span class="van-button__text">ADD TO CART</span></div>
      </button></div>
    <div><button class="van-button van-button--danger van-button--large">
        <div class="van-button__content"><span class="van-button__text">Buy</span></div>
      </button></div>
  </div>
</div>
`;
